src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerValidator.cs

```diff
@@ -555,4 +556,22 @@ public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
             return errors;
         }
     }
+
+    public class MemoryOptimizedCannotBeEnabledWhenNotSupportedRule : ITableDesignerValidationRule
+    {
+        public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
+        {
+            var table = designer.TableViewModel;
+            var errors = new List<TableDesignerIssue>();
+            if (!designer.IsMemoryOptimizedTableSupported && designer.TableViewModel.IsMemoryOptimized)
+            {
+                errors.Add(new TableDesignerIssue()
+                {
+                    Description = string.Format("Memory-optimized is either not supported or not enabled in this database."),
```

Comment:
1. change the description to: `Memory optimized table is not supported for this database`
2. use the new property that Aditya added to provide a more information link for this issue and get the link to a Microsoft docs website that covers this topic.

Reply:
I'm thinking about adding a `IsAzure` property to `TableDesigner` and link to:
1. https://docs.microsoft.com/en-us/sql/relational-databases/in-memory-oltp/overview-and-usage-scenarios if the server is not Azure
2. https://docs.microsoft.com/en-us/azure/azure-sql/in-memory-oltp-overview if the server is Azure.

Reply:
marking this as draft for now as `IsAzure` needs DacFx change.